File: flowboard/src/app/core/models/order.ts
import { OrderItem } from "./orderitems";

export interface Order {
    id: number;
    userId: number;
    total: number;
    status: 'pending' | 'shipped' | 'delivered' | 'canceled';
    name: unknown;
    surname: unknown;
    email: string;
    address: unknown;
    cap: string;
    city: string;
    items: OrderItem[];
    createdAt: string;
}

File: flowboard-backend_api/app/models/order.rb
class Order < ApplicationRecord
  belongs_to :user
  has_many :order_items, dependent: :destroy

  validates :name, :surname, :email, :address, :cap, :city, presence: true
  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }
  validates :status, presence: true
  validates :total, numericality: { greater_than_or_equal_to: 0 }
end
